Gems/Atom/RHI/DX12/Code/Source/RHI/BufferMemoryAllocator.cpp

```diff
@@ -119,17 +126,23 @@ namespace AZ
             }
             else
             {
+#else
+                AZ_UNUSED(overrideSubAllocAlignment);
+#endif
                 RHI::BufferDescriptor bufferDescriptor;
                 bufferDescriptor.m_byteCount = azlossy_caster(sizeInBytes);
                 bufferDescriptor.m_bindFlags = m_descriptor.m_bindFlags;
                 bufferMemoryView = AllocateUnique(bufferDescriptor);
+#ifndef USE_DX12MA
             }
+#endif
 
             return bufferMemoryView;
         }
 
         void BufferMemoryAllocator::DeAllocate(const BufferMemoryView& memoryView)
         {
+#ifndef USE_DX12MA
```

Comment:
Maybe a better approach could be a different allocator BufferMemoryAllocator for dx12ma in order to avoid embedding ifdef code like this.